idseq_dag/steps/generate_lz4.py

```diff
@@ -0,0 +1,35 @@
+"""Generate lz4 file given input file"""
+import os
+
+import idseq_dag.util.command as command
+import idseq_dag.util.command_patterns as command_patterns
+import idseq_dag.util.log as log
+
+from idseq_dag.engine.pipeline_step import PipelineStep
+
+class PipelineStepGenerateLZ4(PipelineStep):
+
+    def run(self):
+        # lz4 is "scalable with multi-cores CPU" so we let it parallelize
+        # itself. See https://github.com/lz4/lz4 .
+        for file_list in self.input_files:
+            for input_file in file_list:
+                if input_file.endswith(('.gz', '.zip', '.lz4')):
+                    log.write(f'Skipping already-compressed file {input_file}')
```

Comment:
Avoid using `log.write`, because it doesn't generate JSON entries. Use `log_event` or `log_context` instead.

Reply:
Good catch! I copied `log.write` from another pipeline step. Maybe it's time to convert those other call sites or mark them as deprecated?

Reply:
Yes, but unfortunately it seems python doesn't offer a "@deprecated" annotation as part of the language.